src/main/java/org/folio/rest/support/CqlUtil.java

```diff
@@ -0,0 +1,22 @@
+package org.folio.rest.support;
+
+import java.util.regex.Pattern;
+
+public final class CqlUtil {
+  private static final Pattern CQL_MATCHES_ALL = Pattern.compile(
+      "^ *id *==? *\\* *$|^ *id *==? *\"\\*\" *$|^ *cql.allRecords *= *1 *$");
```

Comment:
Quite ugly to parse CQL that way. It would be better to run the parser and see if returns TRUE (the CQL parser of folio-vertx-lib simply returns null (to indicate no WHERE clause). But since the result of this is for optimization , I guess it doesn't matter.